financial/exponential_moving_average.py

```diff
@@ -0,0 +1,72 @@
+"""
+    Calculates exponential moving average (EMA) on the series of numbers
+    Wikipedia Reference: https://en.wikipedia.org/wiki/Exponential_smoothing
+    Reference: https://www.investopedia.com/terms/e/ema.asp#toc-what-is-an-exponential-moving-average-ema
+
+    Exponential moving average is used in finance to analyze changes stock prices.
+    EMA is used in conjunction with Simple moving average (SMA), EMA reacts to the
+    changes in the value quicker than SMA, which is one of the advantages of using EMA.
+"""
+
+from collections.abc import Iterator
+
+
+def exponential_moving_average(
+    series_generator: Iterator[float], window_size: int
+) -> Iterator[float]:
+    """
+    Returns the generator which generates exponential moving average of the given
+    series generator
+    >>> list(exponential_moving_average((ele for ele in [2, 5, 3, 8.2, 6, 9, 10]), 3))
+    [2.0, 3.5, 3.25, 5.725, 5.8625, 7.43125, 8.715625]
+
+    :param series_generator: Generator which generates numbers
+    :param window_size: Window size for calculating average (window_size > 0)
+    :return: Returns generator of which returns exponentially averaged numbers
+
+    Formula:
+
+    st = alpha * xt + (1 - alpha) * st_prev
```

Comment:
Please work on self-documenting variable names.  Is `st` `smoothing_time`?

Does `st` stand for `start` or `stop` or `static` or `study` or `stationary` or `standard_time`?  Why does the reader have to guess?  What does `xt` stand for?

Reply:
I have added the definition of the terms below the formula.

Reply:
The definition is NOT what we are looking for.  We want variable names that explain what they are so that when a visitor is reading the code, they do not need to look up terms in the translation table.